Windows tool_integration_tests_2_9 is 3.00% flaky

<!-- meta-tags: To be used by the automation script only, DO NOT MODIFY.
{
  "name": "Windows tool_integration_tests_2_9"
}
-->

The post-submit test builder `Windows tool_integration_tests_2_9` had a flaky ratio 3.00% for the past (up to) 100 commits, which is above our 2.00% threshold.

One recent flaky example for a same commit: https://ci.chromium.org/ui/p/flutter/builders/prod/Windows%20tool_integration_tests_2_9/99
Commit: https://github.com/flutter/flutter/commit/566f606824c46c8ed1a4e841ce62c8a8298fe0bf

Flaky builds:
https://ci.chromium.org/ui/p/flutter/builders/prod/Windows%20tool_integration_tests_2_9/99
https://ci.chromium.org/ui/p/flutter/builders/prod/Windows%20tool_integration_tests_2_9/93
https://ci.chromium.org/ui/p/flutter/builders/prod/Windows%20tool_integration_tests_2_9/62

Recent test runs:
https://flutter-dashboard.appspot.com/#/build?taskFilter=Windows%20tool_integration_tests_2_9

Please follow https://github.com/flutter/flutter/blob/master/docs/infra/Reducing-Test-Flakiness.md#fixing-flaky-tests to fix the flakiness and enable the test back after validating the fix (internal dashboard to validate: go/flutter_test_flakiness).

It looks like we're hitting a JVM heap limit when running Gradle:

```
FAILURE: Build failed with an exception.

* What went wrong:
  Execution failed for task ':app:compressDebugAssets'.
  > A failure occurred while executing com.android.build.gradle.internal.tasks.CompressAssetsWorkAction
  > Java heap space
```

--- a/packages/flutter_tools/test/integration.shard/test_data/deferred_components_project.dart
+++ b/packages/flutter_tools/test/integration.shard/test_data/deferred_components_project.dart
@@ -242,7 +242,7 @@ class BasicDeferredComponentsConfig extends DeferredComponentsConfig {
 
   @override
   String get androidGradleProperties => '''
-  org.gradle.jvmargs=-Xmx4G -XX:MaxMetaspaceSize=2G -XX:+HeapDumpOnOutOfMemoryError
+  org.gradle.jvmargs=-Xmx8G -XX:MaxMetaspaceSize=2G -XX:+HeapDumpOnOutOfMemoryError
   android.useAndroidX=true
   android.enableJetifier=true
   android.enableR8=true

